### PR TITLE
[MINOR][PYTHON][DOC] Fixed instructions for building python docs

### PR DESCRIPTION
**Pull request by Transurgeon**
Adding python packages to README and requirements.txt to fix issues with building python documentation

There are currently two ways to install python packages for building python documentation:
1. running sudo pip install from the README.md
```sh
$ sudo pip install 'sphinx<3.1.0' mkdocs numpy pydata_sphinx_theme ipython nbsphinx numpydoc sphinx-plotly-directive 'jinja2<3.0.0'
```
2. running the requirements.txt file, from $SPARK_HOME
```sh
pip install -r dev/requirements.txt
```
### What changes were proposed in this pull request?
1. Adding 'markupsafe==2.0.1' to both /docs/README.md and /dev/requirements.txt
2. Syncing both files to contain the same packages 
3. Adding pyarrow to /docs/README.md 

### Why are the changes needed?
1. markupsafe==2.1.1 would run into errors when building python documentation (using 'make html')
image
[Similar stack overflow issue](https://stackoverflow.com/questions/72191560/importerror-cannot-import-name-soft-unicode-from-markupsafe)
Downgrading to markupsafe==2.0.1 fixes missing 'soft_unicode' module issues
2. Adding missing packages from requirements.txt (Under Documentation-Python) to README.md, this fixes some additional bugs when trying to build docs from the README command.
3. Finally, running the README command runs into one last error:
image
This is because 'pyarrow' module has been added to requirements.txt (Under PySpark Optional Dependencies) but is still missing from the README command

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Documentation changes only.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Thanks for taking a look for this @Transurgeon

File: docs/README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ See also https://issues.apache.org/jira/browse/SPARK-35375.
 -->
 
 ```sh
-$ sudo pip install 'sphinx<3.1.0' mkdocs numpy pydata_sphinx_theme ipython nbsphinx numpydoc sphinx-plotly-directive 'jinja2<3.0.0'
+$ sudo pip install 'sphinx<3.1.0' mkdocs numpy pydata_sphinx_theme ipython nbsphinx numpydoc sphinx-plotly-directive pyarrow 'jinja2<3.0.0' 'markupsafe==2.0.1' 'docutils<0.18.0'
```
Comment (HyukjinKwon) [inline docs/README.md:64]: Maybe let's just fix this to `sudo pip install dev/requirements.txt`

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

File: dev/requirements.txt [hunk 2]
```diff
@@ -36,6 +36,7 @@ jinja2<3.0.0
 sphinx<3.1.0
 sphinx-plotly-directive
 docutils<0.18.0
+markupsafe==2.0.1
```
Comment (HyukjinKwon) [inline dev/requirements.txt:39]: I would add an inlined comment here with linking a JIRA like SPARK-38279

File: docs/README.md (hunk 1 above)
Comment (Transurgeon) [reply docs/README.md:64]: I was thinking about that..

But the requirements.txt makes you download many more packages which aren't specific to the documentation. I feel like it is good as is, we could consider adding a section/pointing to the requirements.txt for an alternative

**Comment by Transurgeon** [comment]
Thanks for looking into my pull request @HyukjinKwon,

Initially I wanted to work on [SPARK-33247](https://issues.apache.org/jira/browse/SPARK-33247), I was wondering if this is still required. If so, do you have any templates I can base out of for the structure of the examples/scenarios?

**Comment by HyukjinKwon** [comment]
That's awesome. We don't have any templates for now. Basic idea is to have Parameters and Examples that are self-contained (just like `pyspark.pandas` packages). feel free to create a subtask and start working on them

File: docs/README.md (hunk 1 above)
Comment (HyukjinKwon) [reply docs/README.md:69]: That's fine. You can just mention like "`dev/requirements.txt` includes all packages for developement so this is the easiest way to install them". (feel free to rephrase it)

File: docs/README.md [hunk 3]
```diff
@@ -59,9 +59,14 @@ See also https://github.com/sphinx-doc/sphinx/issues/7551.
...
 -->
+Run the following command from $SPARK_HOME
+```sh
+$ sudo pip install -r dev/requirements.txt
+```
 
+Alternatively, for packages strictly related to documentation:
 ```sh
-$ sudo pip install 'sphinx<3.1.0' mkdocs numpy pydata_sphinx_theme ipython nbsphinx numpydoc sphinx-plotly-directive 'jinja2<3.0.0'
+$ sudo pip install 'sphinx<3.1.0' mkdocs numpy pydata_sphinx_theme ipython nbsphinx numpydoc sphinx-plotly-directive pyarrow 'jinja2<3.0.0' 'markupsafe==2.0.1' 'docutils<0.18.0'
```
Comment (HyukjinKwon) [inline docs/README.md:69]: Let's just remove this. I think it's fine. We also have some comments in dev/requirements.txt file too.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**